Dockerfile

```diff
@@ -1,249 +1,46 @@
+RUN python3 setup.py bdist_wheel
+RUN pip3 wheel -r /opt/rikai/python/docker-requirements.txt
 
-#RUN python -c "import torch;torch.cuda.get_device_name(0)"
+FROM apache/spark-py:v${SPARK_VERSION} AS jupyter
 
-# ------------------------- 3. Remove intermediates -------------------------
-FROM ubuntu:20.04 AS output
-LABEL maintainer="rikai developers<rikai-dev@eto.ai>"
-LABEL description="Demo Image"
-
-ARG NB_USER="eto"
-ARG NB_UID="1000"
-
-# necessary tools
-ENV DEBIAN_FRONTEND noninteractive
+USER root
 
-RUN apt-get update && apt-get install -qy --no-install-recommends \
-    autoconf \
-    automake \
-    build-essential \
-    ca-certificates \
-    curl \
-    fontconfig \
-    git \
-    jq \
-    libsnappy-dev \
-    libtool \
-    locales \
-    lsof \
-    openjdk-11-jdk \
-    python-dev \
-    sudo \
-    unzip \
-    vim \
-    wget \
-    && \
-    fc-cache -f && \
+RUN apt -y -qq update && \
+    apt install -y -qq sudo curl aria2 && \
     apt-get clean && \
-    rm -rf /var/lib/apt/lists/*
-
-# create new user with sudo permission
-RUN adduser --disabled-password \
-    --gecos "Default user" \
-    --uid ${NB_UID} \
-    ${NB_USER} && \
-    echo "${NB_USER} ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers && \
-    mkdir -p $HOME/.ssh && \
-    rm -rf /opt/conda /opt/npm
-
-ENV SHELL=/bin/bash \
-    NB_USER=$NB_USER \
-    NB_UID=$NB_UID \
-    USER=$NB_USER \
-    HOME=/home/$NB_USER \
-    NPM_DIR=/opt/npm \
-    CONDA_DIR=/opt/conda
-
-ENV NPM_CONFIG_GLOBALCONFIG=${NPM_DIR}/npmrc \
-    NB_PYTHON_PREFIX=$CONDA_DIR \
-    KERNEL_PYTHON_PREFIX=$CONDA_DIR \
-    PATH=$CONDA_DIR/bin:$NPM_DIR/bin:${HOME}/.local/bin:$PATH
-
-COPY --from=builder --chown=eto:eto /opt/npm /opt/npm
-COPY --from=builder --chown=eto:eto /opt/conda /opt/conda
-COPY --from=builder --chown=eto:eto /home/$NB_USER /home/$NB_USER
-
-
-# Spark
-
-COPY --from=builder $SPARK_HOME $SPARK_HOME
+    rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
 
-USER $NB_UID
-ENV SPARK_HOME=/usr/local/spark
-ENV PYTHONPATH="${SPARK_HOME}/python:${SPARK_HOME}/python/lib/py4j-${py4j_version}-src.zip" \
-    SPARK_OPTS="--driver-java-options=-Xms1024M --driver-java-options=-Xmx4096M --driver-java-options=-Dlog4j.logLevel=info" \
-    PATH=$PATH:$SPARK_HOME/bin
+# copy the wheels over and install all of them
+RUN mkdir -p /opt/rikai/wheels
+COPY --from=whl_builder /opt/rikai/python/dist/rikai-*.whl /opt/rikai/wheels/
+COPY --from=whl_builder /opt/rikai/python/*.whl /opt/rikai/wheels/
+RUN pip3 install --no-cache /opt/rikai/wheels/*.whl && \
+    rm -rf /tmp/* /var/tmp/* /opt/rikai/wheels
 
-# ------------------
-# Launch Jupyter Lab
-# ------------------
+# Copy the jar to the class path
+COPY --from=jar_builder /root/.ivy2/local/ai.eto/rikai_2.12/*/jars/rikai_2.12.jar /opt/spark/jars/
 
-WORKDIR ${HOME}/rikai/notebooks
+RUN mkdir -p /opt/rikai/notebooks
+COPY ./notebooks/Coco.ipynb /opt/rikai/notebooks/Coco.ipynb
+COPY ./notebooks/Mojito.ipynb /opt/rikai/notebooks/Mojito.ipynb
```

Comment:
remove it now or next time

I will change the link to the google lab, because we will not support google colab in later rikai release due to the python 3.7 problem.